fdbserver/workloads/RemoveServersSafely.actor.cpp

```diff
@@ -452,18 +452,35 @@ struct RemoveServersSafelyWorkload : TestWorkload {
 		// Swap coordinator with one server in the kill set to ensure the number of processes to kill does not increase.
 		// This is needed only if a new coordinator is added to the toKill set in this function and safety check passes
 		if (markExcludeAsFailed && coordExcl.isValid()) {
+			// Situation where the entirety of original kill set is selected and extra coordinator is added
+			// Shrink down failed vector to maintain size guarantees
+			if (toKillMarkFailedArray.size() > toKillArray.size()) {
+				auto removeServer = toKillMarkFailedArray.begin();
+				TraceEvent("RemoveAndKill", functionId)
+				    .detail("Step", "ShrinkFailedKillSet")
+				    .detail("Removing", removeServer->toString());
+				toKillMarkFailedArray.erase(removeServer);
+			}
```

Comment:
Let's add an ASSERT here `toKillMarkFailedArray.size() <= toKillArray.size()` just in case.